drivers/wifi/simplelink/simplelink_sockets.c

```diff
@@ -541,7 +541,7 @@ static int simplelink_poll(struct pollfd *fds, int nfds, int msecs)
 	}
 
 	/* Convert time to SlTimeval struct values: */
-	if (msecs == K_FOREVER) {
+	if (msecs == SYS_FOREVER_MS) {
```

Comment:
@vanti and @pfalcon is this acceptable in a `_poll()` implementation, or should I hardcode `-1` instead if that is what POSIX says?

Reply:
I think it should be -1, yes. We should not mix domains of Zephyr kernel API and POSIX API, this way first can be adjusted without affecting use of 2nd.

Reply:
From what I can tell this is from the fdtable abstraction, and supports "any I/O object implementing POSIX I/O semantics".  So -1 seems better.

BTW @carlescufi the analogous code in drivers/wifi/eswifi/eswifi_socket_offload.c and drivers/modem/modem_socket.c needs to be converted too, if that isn't being done somewhere else.  Both appear to pass an integer `msecs` to `k_sem_take`.

Reply:
Thanks @carlescufi for the looking into this. I agree with others that it should be -1. In fact, z_impl_zsock_poll() should probably have used -1 as well in case the value of SYS_FOREVER_MS changes, but that would belong in a different PR.

I now see that the remaining build errors are coming from posix.